sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/GetTypeWithIsAssignableFrom.CSharp9.cs

```diff
@@ -0,0 +1,15 @@
+﻿Apple apple = new();
+Fruit f = apple;
+
+var b = f is Apple; // Compliant
+b = f is not Apple; // Compliant
+b = apple.GetType() == typeof(int?); // Compliant
+b = apple.GetType().IsInstanceOfType(f.GetType()); // Compliant
+b = f as Apple == null; // Noncompliant
+b = f as Apple is null; // Compliant - FN
+b = f as Apple is not null; // Compliant - FN
```

Comment:
It's confusing. Not sure if it comply to RSPEC or not
```suggestion
b = f as Apple is null; // FN
b = f as Apple is not null; // FN
```

Reply:
> It's confusing. Not sure if it comply to RSPEC or not

I also find these confusing and that's the reason I've marked them as false negatives. [Rspec](https://rules.sonarsource.com/csharp/RSPEC-2219) needs to be updated a bit with new examples but I think raising on these is still in the rule scope.